### PR TITLE
Eddie/valay/refactor 

### PR DESCRIPTION
**Pull request by emattia**
@valayDave main structural update is that to run `@batch`, I reinserted the `update_ssh_config_and_restart_sshd` function in `mpi_setup.py`. For now this unblocks using the same code path on AWS Batch and Kubernetes. Without that function, `@batch` produces:
```
Failed to capture the SSH keys using `ssh-keyscan` for the worker type 0 and node index <WORKER IP>. `ssh-keyscan` failures indicate that the nodes are unable to communicate with each other via ssh. 
```

FYI for testing: running with GPUs on batch requires setting `@environment(vars={"NCCL_SOCKET_IFNAME": "eth0"})`.

All other updates we discussed offline are applied in this PR.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 from setuptools import setup, find_namespace_packages
 
-version = "0.0.3"
```
Comment (valayDave) [inline setup.py:3]: why bump down the version ?

File: metaflow_extensions/deepspeed/plugins/deepspeed_libs/deepspeed_decorator.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+                # try:
+                #     self.n_slots = int(compute_deco_attrs["gpu"])
+                #     self.is_gpu = True
+                # except KeyError:
+                #     self.n_slots = int(compute_deco_attrs["cpu"])
+                #     self.is_gpu = False
+                # if not self.n_slots > 0:
+                #     self.n_slots = int(compute_deco_attrs["cpu"])
+                #     self.is_gpu = False
+                # break
```
Comment (valayDave) [inline metaflow_extensions/deepspeed/plugins/deepspeed_libs/deepspeed_decorator.py]: remove comment here.

File: metaflow_extensions/deepspeed/plugins/deepspeed_libs/deepspeed_decorator.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        from metaflow.plugins.aws.aws_utils import compute_resource_attributes
+
+        self.environment = environment
+        self.flow_datastore = flow_datastore
+
+        for deco in decos: 
+            if deco.name in ["resources", "kubernetes", "batch"]:
+                if deco.attributes['gpu']:
+                    self.is_gpu = True
+                    self.n_slots = deco.attributes['gpu']
```
Comment (valayDave) [inline metaflow_extensions/deepspeed/plugins/deepspeed_libs/deepspeed_decorator.py:61]: TODO For the future. `n_slots` can be configurable ?

Comment (emattia) [reply metaflow_extensions/deepspeed/plugins/deepspeed_libs/deepspeed_decorator.py:61]: Yeah, overriding that part of hostfile is straightforward. Would you want to set this as an arg to `@deepspeed`, or in the `current.deepspeed.run` args?

File: metaflow_extensions/deepspeed/plugins/deepspeed_libs/deepspeed_decorator.py (hunk 2 above)
Comment (emattia) [reply metaflow_extensions/deepspeed/plugins/deepspeed_libs/deepspeed_decorator.py]: oops, dropping this.

File: setup.py (hunk 1 above)
Comment (emattia) [reply setup.py:3]: It was a mistake having it at `0.0.5`, the latest pypi release is `0.0.3`.

**Pull request merged**